onnxruntime/core/optimizer/transpose_optimizer/layout_tranformer_dev_notes.md

```diff
@@ -0,0 +1,53 @@
+
+### Convert layout for applicable nodes
+[This is the first pass](https://github.com/microsoft/onnxruntime/blob/1a4868e5c4c4a270ad91036e36f2a03410c4c278/onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc#L815). Layout transformer simply inserts transpose nodes before and after every layout sensitive ops that are claimed by the EP. After this pass, the graph is correct but extremely inefficient.
+
+### Optimize the converted graph
+After the first pass is complete, layout transformer calls the transpose optimizer to remove all the canceling as well as redundant transposes from the graph. The following passes happen as part of transpose optimization.
+
+1. Iterate over sorted nodes in reverse order to find which outputs have paths through supported ops to  transpose nodes. Transposes are pulled pushed towards these outputs. Graph is not altered in this pass. [Code](https://github.com/microsoft/onnxruntime/blob/1a4868e5c4c4a270ad91036e36f2a03410c4c278/onnxruntime/core/optimizer/transpose_optimizer/transpose_optimizer.cc#L1875)
+
+2. Push transposes through applicable nodes and remove canceling transposes. At the end of this pass the model will be efficient and will only contain the transpose ops which are necessary for correctness. [Code](https://github.com/microsoft/onnxruntime/blob/1a4868e5c4c4a270ad91036e36f2a03410c4c278/onnxruntime/core/optimizer/transpose_optimizer/transpose_optimizer.cc#L1905)
+
+3. This is the final optimization pass. In this pass, any transpose node (not part of a QDQ group) if succeeds a DQ node, is moved above the DQ node. In QDQ models this helps to preserve the QDQ node group when a Transpose was pushed across a DQ into an existing QDQ node group. In all other scenarios this is beneficial as well because moving transpose above DQ node is more efficient as transpose node now handles less data. [Code](https://github.com/microsoft/onnxruntime/blob/1a4868e5c4c4a270ad91036e36f2a03410c4c278/onnxruntime/core/optimizer/transpose_optimizer/transpose_optimizer.cc#L1956)
+
+## Enabling layout transformer for a new Compile based EP
+For details please refer to [PR 10371](https://github.com/microsoft/onnxruntime/pull/10371) which introduced layout transformer and enabled it for NNAPI.
+
+Basic steps are as follows:
+1. Implement [GetPreferredLayout](https://github.com/microsoft/onnxruntime/blob/1a4868e5c4c4a270ad91036e36f2a03410c4c278/include/onnxruntime/core/framework/execution_provider.h#L285) method for the EP which overrides the base class method.
+2. Remove any existing logic in the EP to convert layouts
+3. Add a validation method similar to [IsOpInRequiredLayout](https://github.com/microsoft/onnxruntime/blob/1a4868e5c4c4a270ad91036e36f2a03410c4c278/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc#L502) to validate the ops are in required layout.
```

Comment:
Maybe mention that this only applies to layout sensitive ops, and the validation is a check that the domain kMSInternalNHWCDomain